libs/openFrameworks/math/ofMatrix4x4.h

```diff
@@ -48,91 +63,145 @@ class ofMatrix4x4 {
 	ofMatrix4x4( const ofMatrix4x4& mat) {
 		set(mat.getPtr());
 	}
+
+	/// \brief You can pass a pointer to floats, and the first 16 contents will be extracted into this matrix.
+	/// Note: the validity of these values is not checked!
 	ofMatrix4x4( float const * const ptr ) {
 		set(ptr);
 	}
+
+	/// \brief Rotation matrices can be constructed from a quaternion
 	ofMatrix4x4( const ofQuaternion& quat ) {
 		makeRotationMatrix(quat);
 	}
 
+	/// \brief You can pass all 16 values of the matrix as positional arguments (in row-major order)
 	ofMatrix4x4(	float a00, float a01, float a02, float a03,
 	              float a10, float a11, float a12, float a13,
 	              float a20, float a21, float a22, float a23,
 	              float a30, float a31, float a32, float a33);
 
-	// \brief destructor
+	/// \brief destructor
 	~ofMatrix4x4() {}
 
-	// Init matrix as identity, scale, translation...
-	// all make* methods delete the current data
+	/// Init matrix as identity, scale, translation...
+	/// all make* methods delete the current data
 	
+	/// \brief Matrix becomes the identity matrix
 	void makeIdentityMatrix();
 
+	/// \brief Matrix becomes a scale matrix with values only in the diagonal.
+	/// Accepts x, y, z scale values as a vector or separately
 	void makeScaleMatrix( const ofVec3f& );
 	void makeScaleMatrix( float, float, float );
 
+	/// \brief Matrix becomes a translation matrix with values only in the fourth column.
+	/// Accepts x, y, z translation values as a vector or separately
 	void makeTranslationMatrix( const ofVec3f& );
 	void makeTranslationMatrix( float, float, float );
 
+	/// \brief Matrix becomes a rotation matrix with values in the first three rows and columns.
+	/// \brief Matrix becomes a rotation from the first vector direction to the second
 	void makeRotationMatrix( const ofVec3f& from, const ofVec3f& to );
+	/// \brief Matrix becomes a rotation of angle (degrees) around the vector axis.
 	void makeRotationMatrix( float angle, const ofVec3f& axis );
+	/// \brief Matrix becomes a rotation of angle (degrees) around the axis specified positionally
 	void makeRotationMatrix( float angle, float x, float y, float z );
+	/// \brief Matrix becomes a rotation that matches the quaternion's orientation.
 	void makeRotationMatrix( const ofQuaternion& );
+	/// \brief Matrix becomes a rotation that is the result of computing a rotation
+	/// for each of the provided axes, in order of parameters.
 	void makeRotationMatrix( float angle1, const ofVec3f& axis1,
 	                 float angle2, const ofVec3f& axis2,
 	                 float angle3, const ofVec3f& axis3);
 
 
-	// init related to another matrix
+	/// init related to another matrix
+	/// \brief Matrix becomes the inverse of the provided matrix.
 	bool makeInvertOf( const ofMatrix4x4& rhs);
+	/// \brief Matrix becomes an orthonormalized version of the provided matrix.
+	/// The basis vectors (the 3x3 chunk embedded in the upper left of the matrix)
```

Comment:
For some of these more complicated mathematical concepts, it would be totally acceptable and helpful to give a link to the relevant wikipedia article.  See other examples in [ofMath](https://github.com/openframeworks/openFrameworks/blob/master/libs/openFrameworks/math/ofMath.h#L216).